dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/ItemRestRepository.java

```diff
@@ -46,21 +48,24 @@ public ItemRestRepository() {
     }
 
     @Override
-    public ItemRest findOne(Context context, UUID id) {
+    @PreAuthorize("hasPermission(#id, 'ITEM', 'READ')")
+    public ItemRest findOne(UUID id) {
         Item item = null;
         try {
-            item = is.find(context, id);
+            item = is.find(obtainContext(), id);
         } catch (SQLException e) {
             throw new RuntimeException(e.getMessage(), e);
         }
         if (item == null) {
-            return null;
+            throw new ResourceNotFoundException();
         }
         return converter.fromModel(item);
     }
 
     @Override
-    public Page<ItemRest> findAll(Context context, Pageable pageable) {
+    @PreAuthorize("hasAuthority('ADMIN')")
```

Comment:
Is findAll() a method that should only be accessed by an Admin?  Does findAll() ever mean "find all that I can access" or is that type of operation handled elsewhere?

Reply:
Currently the `findAll` method returns all items ignoring any resource policies. That is why only admins should be allowed to use this endpoint. (see the discussion in https://github.com/DSpace/DSpace/pull/1879)